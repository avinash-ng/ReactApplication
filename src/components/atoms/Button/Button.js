import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
