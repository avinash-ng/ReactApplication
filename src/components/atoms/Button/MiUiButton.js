import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton