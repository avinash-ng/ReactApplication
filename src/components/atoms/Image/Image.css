img {
    width: 450px;
    height: 450px;
    float: left;
    margin-left: 20px;
    margin-top: 0px;
    padding-top: 60px;
    /* margin-bottom: 30px; */
}