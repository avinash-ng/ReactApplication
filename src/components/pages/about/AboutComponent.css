/* .entire-content {
    margin-left: 500px;
    width: 500px;
    height: 700px;
}

.img-div {
    padding-top: 0px;
}

.entire-content img {
    width: 300px;
    height: 300px;
    margin-top: -50px;
    margin-left: 100px;
}

.entire-content .fname {
    margin-top: 275px;
} */


/* .entire-component .image {
    width: 100px;
    height: 100px;
    border-radius: 50px;
} */


/* .entire-content .image {
    margin-left: 0px;
    width: 257px;
    height: 275px;
}

img {
    padding-top: 0px;
    width: 250px;
    height: 250px;
    border-radius: 50%;
    margin-left: 600px;
}

h3 {
    display: inline;
}

.fname {
    margin-top: 10px;
}
 */


/*use-itty-bitty-template*/


/* //recently commented */


/* body {
    background: #ececec;
}

.entire-content h1,
h2,
h3,
.entire-content h4,
h5 {
    border-bottom: 1px solid #ccc;
    color: #3F51B5;
    padding-bottom: 8px
}

.container {
    margin: auto;
    width: 350px;
    text-align: center;
}

.listFlex {
    display: flex;
    justify-content: center;
}

img {
    padding-top: 0px;
    width: 300px;
    height: 300px;
    border-radius: 50%;
    margin-bottom: 10px;
    ;
}

p {
    font-size: 15px;
}

.entire-content h4 {
    font-size: 18px;
} */

.card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    max-width: 300px;
    margin-left: 100px;
    text-align: center;
    font-family: arial;
}

.title {
    color: grey;
    font-size: 18px;
}

button {
    border: none;
    outline: 0;
    display: inline-block;
    padding: 8px;
    color: white;
    background-color: #000;
    text-align: center;
    cursor: pointer;
    width: 100%;
    font-size: 18px;
}

.entire-content a {
    text-decoration: none;
    font-size: 22px;
    color: black;
}

.entire-content button:hover,
.entire-content a:hover {
    opacity: 0.7;
}

.card img {
    margin-left: 0px;
    width: 300px;
    height: 250px;
    padding-top: 0px;
}

.brief-content {
    margin-top: -500px;
    margin-left: 500px;
    width: 750px;
}

.brief-content span {
    font-size: 20px;
}


/* .entire-content {
    margin-top: 0px;
}*/

.card {
    float: none;
}