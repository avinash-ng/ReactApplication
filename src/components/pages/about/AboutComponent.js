import React, { Fragment }  from 'react';
import DrawerComponent from "../../template/pagestemplate/DrawerComponent"
import { useAuth0 } from '../../../react-auth0-spa';
import ProfilePic from '../../atoms/ProfilePicture/ProfilePic';
import Label from '../../atoms/Label/Label';
// import './AboutComponent.css'
import history from '../../../utils/history'
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
    briefcontent: {
        marginTop: '-500px',
        marginLeft: '500px',
        width: '750px',
    },
    span: {
        fontSize: '20px'
    },
    image: {
        marginLeft: '0px',
        width: '300px',
        height: '250px',
        paddingTop: '0px'
    },
    button: {
        border: 'none',
        outline: '0',
        display: 'inline - block',
        padding: '8px',
        color: 'white',
        backgroundColor: '#000',
        textAlign: 'center',
        cursor: 'pointer',
        width: '300px',
        fontSize: '18px'
    },
    title: {
        color: 'grey',
        fontSize: '18px',
    },
    card: {
        boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.2)',
        maxWidth: '300px',
        marginLeft: '100px',
        textAlign: 'center',
        fontFamily: 'arial',
    },
    anchor: {
        textDecoration: 'none',
        fontSize: '22px',
        color: 'black'
    }
}))

const ContentComponent = () => {
    const classes = useStyles();
    const { loading } = useAuth0() || true;
    const {user} = useAuth0() || {name: 'avinash'}

    if (loading || !user) {
        return <div>Loading...</div>;
    }
    const handleClick = () => {
        history.push('/contact')
    }

    return (
        <div className="entire-content">



            <div className="card" className={classes.card}>
                <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></link>
                <img className={classes.image} src={user.picture} alt="John" />
                <h1>{user.name}</h1>
                <p className="title" className={classes.title}>Student</p>
                <p style={{fontSize: '18px'}}>Vishnu Institute Of Technology</p> 
                <div style={{ margin: '24px 0' }}>
                    <Link className={classes.anchor} to="#" style={{marginRight: '10px'}}><i className="fa fa-dribbble"></i></Link> 
                    <Link className={classes.anchor} to="#" style={{marginRight: '10px'}}><i className="fa fa-twitter"></i></Link>  
                    <Link className={classes.anchor} to="www.facebook.com" style={{marginRight: '10px'}}><i className="fa fa-linkedin"></i></Link>  
                    <Link className={classes.anchor} to="#"><i className="fa fa-facebook"></i></Link>  
                </div>
                <button style={{ marginLeft: '0px' }} className={classes.button} onClick={handleClick}>Contact</button>
            </div>

            <div className="brief-content" className={classes.briefcontent}>
                <h1>Brief About Me</h1>
                <span classname={classes.span}>Hi i am Avinash Kolaparthi. 
                
                </span>
                <div className="place-info">
                    <h2>Where do i live </h2>
                    <span classname={classes.span}>I am living in Bhimavaram.</span>
                </div>
                <div className="study-info">
                    <h2>Study Info</h2>
                    <span classname={classes.span}>I am in final year final semester bachelor degree studying in vishnu institute of Technology.
                    </span>
                </div>
            </div>

        </div>
    );
}
const AboutComponent = () => {
    
    return (
        
        <DrawerComponent > <ContentComponent /> </DrawerComponent>   
            
        
    )
}
export default AboutComponent;