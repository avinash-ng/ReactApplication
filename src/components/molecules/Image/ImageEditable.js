import React from 'react'
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from "../../atoms/Button/Button"
const ImageEditable = props => {
    return (
        <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}>
            {/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */}
            <ImageTag className={props.className} onClicked={props.headingOnClicked} object={props.object}></ImageTag>
            {props.object.selected ? <Button buttonName={props.buttonName} onClicked={props.buttonOnClicked}></Button> : ""}

        </div>
    )
}
export default ImageEditable