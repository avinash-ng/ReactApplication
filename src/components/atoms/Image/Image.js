import React from 'react'
import PropTypes from "prop-types"
import "./Image.css"
const Image = (props) => {
    return (
        <img data-testid="image" src={props.imageName} alt={props.altText} />
    )
}
Image.propTypes = {
    imageName: PropTypes.string.isRequired,
    altText: PropTypes.string.isRequired
} 

export default Image