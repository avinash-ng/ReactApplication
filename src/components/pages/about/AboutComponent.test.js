const { shallow, configure, mount } = require("enzyme")
import AboutComponent from "./AboutComponent"
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../atoms/Button/Button";
import sinon from 'sinon'
import { render, fireEvent } from "@testing-library/react";
import Image from "../../atoms/Image/Image";
import ImageEditable from "../../molecules/Image/ImageEditable";
import { ListItem, AppBar } from '@material-ui/core'
import {useAuth0} from '../../../react-auth0-spa'
import { TabPanel } from "./AboutComponent";

configure({ adapter: new Adapter() });
const user = {
    email: "av@gmail.com",
    email_verified: true,
    sub: "google-oauth2|12345678901234"
}
jest.mock("../../../react-auth0-spa");
describe('testing <AboutComponent /> ', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            user,
            logout: jest.fn(),
            loginWithRedirect: jest.fn()
        });
    });

    it('renders <img /> correctly', () => {

        const RenderComponent = mount(<AboutComponent > </AboutComponent>)
        expect(RenderComponent.find('img').exists()).toBe(true)
    })

    it('renders div with className brief-content correctly', () => {
        const RenderComponent = mount(<AboutComponent > </AboutComponent>)
        expect(RenderComponent.find('.brief-content').exists()).toBe(true)
    })



    it('matches snapshot', () => {
        const RenderComponent = shallow(<AboutComponent ></AboutComponent>)
        expect(RenderComponent).toMatchSnapshot()
    })
})