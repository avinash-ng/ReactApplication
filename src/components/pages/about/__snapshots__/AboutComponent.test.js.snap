// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <AboutComponent />  matches snapshot 1`] = `ShallowWrapper {}`;
